contrib/inventory/ec2.py

```diff
@@ -338,8 +338,8 @@ def read_settings(self):
         else:
             self.all_elasticache_nodes = False
 
-        # boto configuration profile (prefer CLI argument)
-        self.boto_profile = self.args.boto_profile
+        # boto configuration profile (prefer environment variables then CLI argument)
+        self.boto_profile = os.environ.get('AWS_PROFILE', self.args.boto_profile)
```

Comment:
This is still flipped, this will grab the environment var and use it over something a user specifies on the CLI. To pick CLI first, try `self.boto_profile = self.args.boto_profile or os.environ.get('AWS_PROFILE')` then if that fails, the below conditional will try to use the config file.